res/controllers/Reloop-Digital-Jockey-2-IE.scripts.js

```diff
@@ -0,0 +1,876 @@
+    loopactive: [0x12, 0x4E],
+    fx1assign: [0x27, 0x68],    //this is the shifted Activate 3 button
+    fx2assign: [0x2C, 0x63],    //this is the shifted FX ON button
+    highkill: [0x14, 0x50],
+};
+
+RDJ2.KNOBMAP_CH0_CH1 = {
+    loopSize: [0x28, 0x63],     //this is the shifted Dry/Wet Knob
+};
+
+
+////////////////////////////////////////////////////////////////////////
+// Logging functions                                                  //
+////////////////////////////////////////////////////////////////////////
+
+RDJ2.logDebug = function(msg) {
+    if (RDJ2.debug) {
+        print("[" + RDJ2.id + " DEBUG] " + msg);
+    }
+};
+
+RDJ2.logInfo = function(msg) {
+    print("[" + RDJ2.id + " INFO] " + msg);
+};
+
+RDJ2.logWarning = function(msg) {
+    print("[" + RDJ2.id + " WARNING] " + msg);
+};
+
+RDJ2.logError = function(msg) {
+    print("[" + RDJ2.id + " ERROR] " + msg);
+};
+
+
+////////////////////////////////////////////////////////////////////////
+// Buttons                                                            //
+////////////////////////////////////////////////////////////////////////
+
+RDJ2.MIDI_ON = 0x7F;
+RDJ2.MIDI_OFF = 0x00;
+
+RDJ2.isButtonPressed = function(midiValue) {
+    switch (midiValue) {
+    case RDJ2.MIDI_ON:
+        return true;
+    case RDJ2.MIDI_OFF:
+        return false;
+    default:
+        RDJ2.logError("Unexpected MIDI button value: " + midiValue);
+        return undefined;
+    }
+};
+
+/* Custom buttons */
+
+RDJ2.ShiftButton = function(options) {
+    this.state = false;
+    this.connectedContainers = [];
+    components.Button.call(this, options);
+};
+RDJ2.ShiftButton.prototype = new components.Button({
+    input: function(channel, control, value) {
+        //update shift state
+        this.state = RDJ2.isButtonPressed(value);
+        this.send(this.outValueScale(this.state));
+
+        //call shift()/unshift() for each connected container
+        if (this.state) {
+            this.connectedContainers.forEach(function(container) {
+                container.shift();
+            });
+        } else {
+            this.connectedContainers.forEach(function(container) {
+                container.unshift();
+            });
+        }
+    },
+    isActive: function() {
+        return this.state;
+    },
+    connectContainer: function(container) {
+        if (container instanceof components.ComponentContainer === false) {
+            RDJ2.logError("Container type mismatch");
+        } else {
+            this.connectedContainers.push(container);
+            RDJ2.logDebug("Connected container " + this.connectedContainers.indexOf(container) + " to shift button 0x" + this.midi[1].toString(16));
+        }
+    }
+});
+
+RDJ2.LoopInButton = function(options) {
+    components.Button.call(this, options);
+};
+RDJ2.LoopInButton.prototype = new components.Button({
+    outKey: "loop_start_position",
+    outValueScale: function(value) { return (value >= 0) * this.max; },
```

Comment:
```suggestion
    outValueScale: function(value) { return value >= 0 ? this.on : this.off; },
```

Reply:
Done in the latest commit. Also done for LoopOutButton.